query/src/lib/query-client.ts

```diff
@@ -58,3 +69,361 @@ export function provideQueryClient(queryClient: _QueryClient): Provider {
 export function injectQueryClient() {
   return inject(QueryClientService);
 }
+
+class QueryClient extends _QueryClient {
+  #injector = inject(Injector);
+
+  /**
+   *
+   * Asynchronous function that can be used to get an existing query's cached data.
+   * If the query does not exist, queryClient.fetchQuery will be called and its results
+   * returned.
+   *
+   * @example
+   *
+   * queryClient = injectQueryClient();
+   *
+   * const data = await queryClient.ensureQueryData({ queryKey, queryFn })
+   *
+   */
+  override ensureQueryData<
```

Comment:
Can we please add a basic test type for each method? Just to see that it compiles as expected. See example [here](https://github.com/ngneat/query/blob/main/query/src/tests/query.spec.ts#L63).